File: web/src/api/shopInventoryAlert.js
import service from '@/utils/request'

export const getAlertList = (params) => {
  return service({ url: '/shopInventoryAlert/getAlertList', method: 'get', params })
}

export const handleAlert = (data) => {
  return service({ url: '/shopInventoryAlert/handleAlert', method: 'put', data })
}

export const getSetting = (params) => {
  return service({ url: '/shopInventoryAlert/getSetting', method: 'get', params })
}

export const updateSetting = (data) => {
  return service({ url: '/shopInventoryAlert/updateSetting', method: 'put', data })
}

File: server/router/shop/shop_inventory_alert.go
package shop

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

type ShopInventoryAlertRouter struct{}

func (s *ShopInventoryAlertRouter) InitShopInventoryAlertRouter(Router *gin.RouterGroup) {
	shopAlertRouter := Router.Group("shopInventoryAlert")
	var api = v1.ApiGroupApp.ShopApiGroup.ShopInventoryAlertApi
	{
		shopAlertRouter.GET("getAlertList", api.GetAlertList)
		shopAlertRouter.PUT("handleAlert", api.HandleAlert)
		shopAlertRouter.GET("getSetting", api.GetSetting)
		shopAlertRouter.PUT("updateSetting", api.UpdateSetting)
	}
}
